parlai/core/torch_ranker_agent.py

```diff
@@ -446,6 +488,10 @@ def eval_step(self, batch):
             cand_preds = self.block_repeats(cand_preds)
 
         preds = [cand_preds[i][0] for i in range(batchsize)]
+
+        # clean up candidates
+        if self.eval_candidates == 'fixed':
+            self.fixed_candidates = self.fixed_candidates[: self.num_fixed_candidates]
```

Comment:
question, do we really need to track the number of fixed candidates here? aren't you always adding exactly one candidate here?

Reply:
it's possible that with a larger eval-batchsize, one could be adding more than one candidate; this just ensures that we don't assume it's always exactly 1

Reply:
Sounds like a comment could help

Reply:
i moved this functionality into the new class btw